kinde_sdk/management/management_token_manager.py

```
@@ -270,4 +270,56 @@
+            
+            # Set the validated token
+            token_data = {
+                "access_token": bearer_token,
+                "expires_in": introspection_result.get("exp", int(time.time()) + 3600) - int(time.time())
+            }
+            self.set_tokens(token_data)
+            
+            return introspection_result
+            
+        except requests.exceptions.Timeout:
+            raise Exception(f"Introspection request timed out after 30 seconds for domain {self.domain}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)